core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraUseCase.kt

```diff
@@ -744,7 +747,12 @@ constructor(
                 MediaStoreOutputOptions.Builder(
                     application.contentResolver,
                     MediaStore.Video.Media.EXTERNAL_CONTENT_URI
-                )
+                ).apply {
+                    // apply duration limit if applicable
+                    if (currentSettings.value?.maxVideoDurationMillis != -1L) {
```

Comment:
nit: if `currentSettings.value` is `null`, then it won't be equal to `-1L`, which will trigger a `NullPointerException` where it calls `currentSettings.value!!` below.